design/src/glossary.md

```diff
@@ -255,23 +541,55 @@ In some games there are also many other entities to guide the behavior of large
 
 ### Undo
 
+Reverse previous directions or actions.
+Ideally combined with a **redo** functionality.
+
+If buildings exist / don't exist when they need to, **marks structures for deletion**, or creates [ghosts](glossary.md#blueprint) in their place.
+
 ## Ecology
 
 These terms come from the science of ecology and its related fields.
 
-### Clay
-
 ### Nutrient cycle
 
-### Sand
-
-### Silt
+The recycling flow of nutrients between organisms, decaying matter and inorganic stores.
 
 ### Soil
 
+Dirt, but viewed scientifically.
+Useful for growing stuff.
+
+Soil has four main components, which vary in [water storage capacity](glossary.md#water-storage-capacity) and [drainage](glossary.md#water-drainage).
+
+1. Sand: large particles. Drains well, but water storage capacity suffers.
+2. Silt: medium particles. Pretty balanced.
+3. Clay: small particles. Drains terrible, but with great water storage capacity.
+4. Organic matter: used to be living stuff, carefully prepared and mixed in the soil. Great drainage and water storage, but decomposes.
```

Comment:
Is it possible for something to be great at draining, but also be great at water storage?

Also, stuff like peat (or general bog-like stuff) is quite "clay-like" in behaviour but very organic?

I am not sure if organic matter should be a separate category, rather each one of sand/silt/clay can be made up, to varying degrees, of organic matter?